doc/design/resource-dependency.md

```diff
@@ -0,0 +1,343 @@
+
+    ```go
+    type DataInputs []DataInput
+
+    type DataInput struct {
+      // FromDataOutput specifies the name of the DataOutput object in the same namespace.
+      FromDataOutput string `json:"fromDataOutput"`
+
+      // ToFieldPaths specifies an array of fields within this resource object
+      // that will be overwritten by the value of given DataOutput. The type of the
+      // parameter (e.g. int, string) is inferred from the type of these fields;
+      // All fields must be of the same type. Fields are specified as JSON field
+      // paths without a leading dot, for example 'spec.replicas'.
+      // If empty, it means this input is not used at all.
+      // This is allowed for resource ordering purpose.
+      ToFieldPaths []string `json:"toFieldPaths"`
+    }
+    ```
+
+    For example, a web application may specify input of mysql secret in yaml style:
+
+    ```yaml
+    datainputs:
+    - toFieldPaths: ["spec.connSecret"]
+      fromDataOutput: mysql-conn-secret # check below DataOutput of the same name
+    ```
+
+   The above DataInputs will be json-marshaled and put into an annotation key `core.oam.dev/datainputs` of my-web workload:
+
+    ```yaml
+    kind: MyWorkload
+    metadata:
+      name: my-web
+      annotations:
+        "core.oam.dev/datainputs": "[{\"toFieldPaths\": [\"spec.connSecret\"],\"fromDataOutput\": \"mysql-conn-secret\"}]"
+    ```
+
+    Note that my-web won't be created since it has a DataInputs dependency to satisfy.
+
+2. Add a new CRD `DataOutput` with the following definition:
+
+    ```go
+    type DataOutput struct {
+      metav1.TypeMeta   `json:",inline"`
+      metav1.ObjectMeta `json:"metadata,omitempty"`
+
+      Spec   DataOutputSpec   `json:"spec,omitempty"`
+      Status DataOutputStatus `json:"status,omitempty"`
+    }
+
+    type DataOutputSpec struct {
+      // Resource specifies the reference of specific resource, which points to a unique resource.
+      Resource ResourceReference `json:"resource"`
+
+      // FieldPath specifies the field path of a given k8s-style resource.
+      FieldPath string `json:"fieldPath"`
+    }
+
+    type ResourceReference struct {
+      APIVersion string `json:"apiVersion"`
+
+      Kind string `json:"kind"`
+
+      // Name indicates the name of the object in the same namespace.
+      Name string `json:"name,omitempty"`
+    }
+
+    type DataOutputStatus struct {
+      // Ready indicates whether the field data of the source resource is ready.
+      Ready bool `json:"ready"`
+    }
+    ```
+
+    For example, a mysql connection secret DataOutput coming from RDSInstance my-rds would look like:
+
+    ```yaml
+    apiVersion: experimental.core.oam.dev/v1alpha2
+    kind: DataOutput
+    metadata:
+      name: mysql-conn-secret
+    spec:
+      resource:
+        apiVersion: alibaba.oam.crossplane.io/v1alpha1
+        kind: RDSInstance
+        name: my-rds
+      fieldPath: "status.connSecret"
```

Comment:
```suggestion
      fieldPath: "spec.writeConnectionSecretToRef"
```
https://doc.crds.dev/github.com/crossplane/provider-alibaba/config/crd/database.alibaba.crossplane.io_rdsinstances.yaml#spec-writeConnectionSecretToRef

Reply:
Well, this is an example to show field passing.
We have many use cases that passes from status, which means they are not available at the beginning.